packages/flutter/lib/src/widgets/value_updater.dart

```diff
@@ -0,0 +1,171 @@
+import 'package:flutter/foundation.dart';
+import 'package:flutter/scheduler.dart';
+
+import 'framework.dart';
+
+/// A builder function used by the [ValueUpdater] widget that takes the current
+/// value and returns a widget.
+typedef ValueUpdaterBuilder<T> = Widget Function(BuildContext context, T value);
+
+/// A widget that updates a value for the widgets built using a builder.
+///
+/// The [ValueUpdater] widget can be used to avoid the need for creating a
+/// stateful widget just to trigger a change in another widget on creation.
+///
+/// If not given a [valueNotifier], will start at the [initialValue] (or the
+/// [initialValue], if given) and toggle to the [value] on the frame after the
+/// widget is first built. It will only toggle once.
+///
+/// If given a [valueNotifier], will start at [initialValue] and
+/// switch to the value of the [valueNotifier], and then continue to switch
+/// to each new value as the [valueNotifier] changes.
+///
+/// Like all stateful widgets in Flutter, if this widget is recreated, it will
+/// lose its current state. To prevent this, give it a [Key] to allow it to be
+/// reused instead of recreated.
+///
+/// {@tool dartpad}
+/// This example shows how to use the [ValueUpdater] widget to trigger an
+/// [AnimatedAlign] animation when a widget is shown for the first time. This is
+/// done without needing to create a stateful widget.
+///
+/// ** See code in examples/api/lib/widgets/value_updater/value_updater.0.dart **
+/// {@end-tool}
+///
+/// {@tool dartpad}
+/// This example shows how to use the [ValueUpdater] widget to trigger an
+/// [AnimatedAlign] animation when a widget is shown for the first time, and
+/// then change the value of the alignment and animate each time a button is
+/// pressed. This is done without needing to create a stateful widget.
+///
+/// ** See code in examples/api/lib/widgets/value_updater/value_updater.1.dart **
+/// {@end-tool}
+
+class ValueUpdater<T> extends StatefulWidget {
```

Comment:
Two similar widgets to this come to mind: We have the `ValueListenableBuilder` (afaik, it does not support a separate `initialValue`) and the `TweenAnimationBuilder`, which allows you to _animate_ from a start to an end value (specified as a Tween). This widget feels like a non-animated version of the `TweenAnimationBuilder`? For the envisioned use cases, could one of these two existing widgets be used instead for the same effect? Or does the `ValueUpdater` further simplify things? Maybe link to these similar widgets from the doc and describe the difference in a short sentence (e.g. in a "see also" section)?